tasks/upgrade.ts

```diff
@@ -1,105 +1,69 @@
-  console.log(`Upgrade InternalMarket ${contracts.internalMarket.address}`);
+  const contractTypeStr = toPascalCase(contractType);
+  const contractFactory = await hre.ethers.getContractFactory(contractTypeStr);
+  const proxyAddress = contracts[contractType].address;
+  console.log(`Upgrade ${contractTypeStr} ${proxyAddress}`);
   console.log("  Network:", hre.network.name);
 
   const answer = await question(
     "This action is irreversible. Please type 'GO' to continue.\n"
   );
 
   if (answer == "GO") {
-    const internalMarketContract = await hre.upgrades.upgradeProxy(
-      contracts.internalMarket.address,
-      internalMarketFactory
+    const contractInstance = await hre.upgrades.upgradeProxy(
+      proxyAddress,
+      contractFactory
     );
-    await internalMarketContract.deployed();
+    await contractInstance.deployed();
 
-    console.log("    Address:", internalMarketContract.address);
-    console.log("InternalMarket upgraded");
+    console.log("    Address:", contractInstance.address);
+    console.log(`${contractTypeStr} upgraded`);
   }
-});
-
-task("upgrade:governance", "Upgrade Governance Token", async (_, hre) => {
-  const governanceTokenFactory = (await hre.ethers.getContractFactory(
-    "GovernanceToken"
-  )) as GovernanceToken__factory;
-
-  const neokingdom = await NeokingdomDAOHardhat.initialize(hre);
-  const contracts = await neokingdom.loadContracts();
-  console.log(`Upgrade GovernanceToken ${contracts.governanceToken.address}`);
-  console.log("  Network:", hre.network.name);
-
-  const answer = await question(
-    "This action is irreversible. Please type 'GO' to continue.\n"
-  );
-
-  if (answer == "GO") {
-    const governanceTokenContract = (await hre.upgrades.upgradeProxy(
-      contracts.governanceToken.address,
-      governanceTokenFactory
-    )) as GovernanceToken;
-    await governanceTokenContract.deployed();
-
-    console.log("    Address:", governanceTokenContract.address);
-    console.log("GovernanceToken upgraded");
-  }
-});
-
-task("impl", "Get Proxy Impl")
-  .addParam("admin", "Proxy Admin")
-  .addParam("address", "Proxy address")
-  .setAction(
-    async ({ admin, address }: { admin: string; address: string }, hre) => {
-      const [deployer] = await hre.ethers.getSigners();
-      const ProxyAdmin = await hre.ethers.getContractFactory("ProxyAdmin");
-      const proxyAdmin = ProxyAdmin.attach(admin).connect(
-        deployer
-      ) as ProxyAdmin;
-
-      console.log("   Proxy Owner:", await proxyAdmin.owner());
-
-      console.log(
-        "    Address:",
-        await proxyAdmin.getProxyImplementation(address)
-      );
+}
+
+task("upgrade")
+  .addPositionalParam("contract", "The smart contract to upgrade")
+  .setAction(async ({ contract }: { contract: string }, hre) => {
+    console.log(contract);
+    if (!validContracts.includes(contract)) {
+      console.error(`Invalid contract. Valid options are: ${validContracts}`);
+      exit();
```

Comment:
`exit()` exits the process with `0`, use `1` to signal that the process exited unsuccessfully.